w08h03/test/pgdp/datastructures/UnitTest.java

```diff
@@ -14,48 +18,88 @@
 import org.junit.jupiter.params.provider.MethodSource;
 
 public class UnitTest {
-	@Test
-	public void emptyTest() {
-		assertThrows(NoSuchElementException.class, () -> (new QuarternarySearchTree<Integer>()).iterator().next());
-	}
+	static int seed = 69420;
 
-	@ParameterizedTest
-	@DisplayName("Dynamic Test")
-	@MethodSource
-	void dynamicTests(String type, int[] expected, int[] input) {
-		QuarternarySearchTree<Integer> tree = new QuarternarySearchTree<Integer>();
+	<Type extends Comparable<Type>> void testArray(Type[] input) {
+		QuarternarySearchTree<Type> tree = new QuarternarySearchTree<Type>();
+		Type[] expected = Arrays.copyOf(input, input.length);
+		Arrays.sort(expected);
 
-		for (int element : input) {
+		for (Type element : input) {
 			tree.insert(element);
 		}
 
 		int position = 0;
-		for (int element : tree) {
-			int exp = expected[position++];
-			assertEquals(exp, element, type + ": Invalid Output at position [" + (position-1) + "]: Expected [" + exp + "], got [" + element + "]");
+		for (Type element : tree) {
+			Type exp = expected[position++];
+			assertEquals(exp, element, "Invalid Output at position [" + (position - 1) + "]: Expected [" + exp
+					+ "], got [" + element + "]");
 		}
 
-		assertEquals(expected.length, position, type + ": Invalid Iteration Count. Expected [" + expected.length + "] got [" + position + "]");
+		assertEquals(expected.length, position,
+				"Invalid Iteration Count. Expected [" + expected.length + "] got [" + position + "]");
+	}
+
+	@Test
+	@DisplayName("should iterate over empty graph")
+	public void emptyTest() {
+		assertThrows(NoSuchElementException.class, () -> (new QuarternarySearchTree<Integer>()).iterator().next());
+	}
+
+	@Test
+	@DisplayName("should iterate over graph in artemis example")
+	public void artemisTest() {
+		testArray(new Integer[] { 8, 4, 12, 1, 5, 9, 13, 3, 7, 11, 15, 2, 6, 10, 14 });
+	}
+
+	@Test
+	@DisplayName("should iterate over a graph of one repeat element")
+	public void oneElementTest() {
+		testArray(new Integer[] { 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1 });
+	}
+
+	@Test
+	@DisplayName("should iterate over a numeric graph")
+	public void numericTest() {
+		IntStream.range(0, 10).forEach(i -> {
+			var args = new Integer[i];
+			var rnd = new Random(seed + i);
+
+			for (int j = 0; j < args.length; j++) {
+				args[j] = rnd.nextInt(Integer.MIN_VALUE, Integer.MAX_VALUE);
+			}
+
+			testArray(args);
+		});
 	}
```

Comment:
there's an ongoing issue with the current generator code, setting range to a higher value such as 100 leads to unexpected behaviour, however this doesn't affect the current PR